GiEnJul/Repositories/CmsRepository.cs

```diff
@@ -0,0 +1,61 @@
+﻿using AutoMapper;
+using GiEnJul.Entities;
+using GiEnJul.Infrastructure;
+using Serilog;
+using System.Collections;
+using System.Threading.Tasks;
+
+namespace GiEnJul.Repositories
+{
+
+    public interface ICmsRepository
+    {
+        Task<IEnumerable> GetCmsByContentType(string contentType);
+        Task<IEnumerable> GetCmsByContentTypeAsync(string contentType);
+        Task<Models.Cms> InsertOrReplaceAsync(Models.Cms cms);
+        Task<IEnumerable> GetSingleCmsByContentTypeAsync(string contentType, string index);
+        Task<IEnumerable> GetSingleCmsByContentTypeAndIndex(string contentType, string index);
+    }
+
+    public class CmsRepository : GenericRepository<Cms>, ICmsRepository
+    {
+        public CmsRepository(ISettings settings, IMapper mapper, ILogger log, string tableName ="Cms") : base(settings, tableName, mapper, log)
+        { }
+
+        public Task<IEnumerable> GetCmsByContentType(string contentType)
+        {
+            var cmsByContentType = GetCmsByContentTypeAsync(contentType);
+
+            return cmsByContentType;
+        }
+
+        public Task<IEnumerable>GetSingleCmsByContentTypeAndIndex(string contentType, string index)
+        {
+            var singleCmsByContentTypeAndIndex = GetSingleCmsByContentTypeAsync(contentType, index);
+            return singleCmsByContentTypeAndIndex;
+        }
```

Comment:
shouldn't return Task if the method is not async.
Are these needed?

Reply:
Agree with Ron!

Reply:
fixed